src/Compilers/CSharp/Test/Emit2/Semantics/InlineArrayTests.cs

```diff
@@ -20154,30 +20154,43 @@ .maxstack  3
             CompileAndVerify(comp, expectedOutput: " 0 1 2 3", verify: Verification.Fails).VerifyDiagnostics();
         }
 
-        [Fact]
+        [ConditionalFact(typeof(CoreClrOnly))]
         public void Foreach_InAsync_03()
         {
             var src = @"
 class Program
 {
-    static async void Test()
+    static Buffer4<int> s_buffer;
+
+    static async System.Threading.Tasks.Task Main()
     {
+        s_buffer[1] = 3;
+
         foreach (ref int y in GetBuffer())
         {
+            y *= y;
+            System.Console.Write(y);
```

Comment:
>System.Console.Write(y); [](http://example.com/codeflow?start=12&length=24)

Please add a test scenario with ```await System.Threading.Tasks.Task.Yield();``` in the loop.

Reply:
Perhaps two separate scenarios with an `await` at the beginning of the loop and an `await` at the end of the loop.

Reply:
Also a flavor with ```GetBuffer()``` captured in a ref local prior to the loop:
```
ref Buffer4<int> buffer = GetBuffer();
foreach (ref int y in buffer)
...
```